[DOC] Documentation model evaluation

maylistran01 opened the pull request:
This PR is for the documentation tasks from Models Evaluation part Léa and I have to do:

- [ ]  Convergence diagnosis

- [x] Fit metrics

- [x] Prediction metrics

maylistran01 committed:
plan proposition for the user guide

maylistran01 committed:
plan proposition for the user guide

maylistran01 committed:
Add metrics part for Lea and Maylis

maylistran01 committed:
Add metrics part for Lea and Maylis

maylistran01 committed:
Equation and bib integration

maylistran01 committed:
Equation and bib integration

JulietteOrtholand commented on docs/models_evaluation.md:
```diff
@@ -1,9 +1,123 @@
+$$
+
+**Package:** `statsmodels.api`
+
```
Finally, you did not talk about the fact that the model is a mixed effect model? You did not cite the paper of [lavielle](https://projecteuclid.org/journals/electronic-journal-of-statistics/volume-8/issue-1/A-note-on-BIC-in-mixed-effects-models/10.1214/14-EJS890.full) on that, maybe could be interesting?

JulietteOrtholand commented on docs/models_evaluation.md:
```diff
@@ -1,9 +1,123 @@
+MAE = \frac{1}{n} \sum_{i=1}^{n} |y_i - \hat{y}_i|
+$$
+
+MAE is more robust to outliers and provides a straightforward sense of typical prediction error.
```
Could you add a small code? + Maybe a reference? Even if it's wikipedia

JulietteOrtholand commented on docs/models_evaluation.md:
```diff
@@ -1,9 +1,123 @@
+#### Residual Q-Q Plot
+A graphical tool to assess whether residuals follow a normal distribution, which is an assumption in many mixed-effects models.\cite{fox2015applied}
+
+**Package:** `statsmodels.api`
```
as before could you put the lik to the documenattion and a small code how to use

JulietteOrtholand commented on docs/models_evaluation.md:
```diff
@@ -1,9 +1,123 @@
+
+In mixed-effects models, multiple R² variants exist (e.g., marginal vs. conditional R²) to account for fixed and random effects. \cite{nakagawa2012method}
+
+**Package:** `sklearn.metrics`
```
code+ link

JulietteOrtholand commented on docs/models_evaluation.md:
```diff
@@ -1,9 +1,123 @@
+#### Integrated Brier Score (IBS)
+The Integrated Brier Score (IBS) is a robust metric used to evaluate the predictive accuracy of survival models. In survival analysis, IBS quantifies how well a model predicts the timing of events by integrating the Brier Score, a measure of prediction error for probabilistic outcomes, across all observed time points. It compares the model’s predicted survival probabilities against actual observe ...
+
+**Package:** `scikit-survival`
```
code + link to the specific function documentation

JulietteOrtholand commented on docs/models_evaluation.md:
```diff
@@ -1,9 +1,123 @@
+#### Cumulative Dynamic AUC
+Cumulative AUC (or time-dependent AUC) evaluates the model’s ability to discriminate between subjects who experience an event by a specific time and those who do not, based on their predicted risk scores. It focuses on ranking accuracy, ensuring high-risk subjects receive higher predicted probabilities than low-risk ones at each time point. Cumulative AUC describes how well risks are ordered. A h ...
+
+**Package:** `scikit-survival`
```
code + link to the specific function documentation

JulietteOrtholand commented on docs/models_evaluation.md:
```diff
@@ -1,9 +1,123 @@
+R² indicates how well the model explains the variance in the observed data. Higher values (closer to 1) suggest better performance.
+
+$$
+R^2 = 1 - \frac{\sum_{i=1}^{n} (y_i - \hat{y}_i)^2}{\sum_{i=1}^{n} (y_i - \bar{y})^2}
```
Maybe if the documentation of the fonction you call is well done, no need for the equations?

maylistran01 replied:
I thought we said in the meeting yesterday that we should add all the equations?

JulietteOrtholand replied:
Yes, you are right, it's just that when reviewing the PR I was wondering how hard it might be to maintain the equations right for a function that is implemented by another package and that might change with time... So in a perfect world I think having the equations in the doc would be nice, but in a realistic world, wouldn't it be better to have no equation but a good reference toward them rather than wrong equation because the function has been updated by the other library... WDYT?

maylistran01 replied:
To what extent do the equations in a package actually change? I would assume—perhaps incorrectly—that formulas like MAE or MSE remain constant. Does this assumption not apply to all the metrics listed here?

JulietteOrtholand replied:
[Here](https://python.arviz.org/en/stable/api/generated/arviz.rhat.html) you have a function to compute a statistic on an MCMC chain. What you can see in the references is that the notion has been "invented" by Gelman and Rubin, yet the equation used is not the one of this version because methodological work has been done since then and a more robust version has been developed. So the package does not change the API but the black box behind to update it. You can also think about it with the conditional and the marginal version of the WAIC. I agree that for the MAE, it might not change so you can keep the equation, but maybe for [Brier score](https://scikit-survival.readthedocs.io/en/stable/api/generated/sksurv.metrics.brier_score.html#sksurv.metrics.brier_score) a new way to compute inverse probability will be used so you might don't want to write it down explicitly... 

Thus, as you point it out, it might depend on the metric, let me know if it is clear!

maylistran01 replied:
Yes it is super clear ! 
Then, would you agree if we keep the equations for the AIC, BIC, MAE, the MSE and the general R2 (used by sklearn_metrics) and not for the other metrics (WAIC, Marginal and conditional R2, Brier score, dynamic AUC)?

JulietteOrtholand commented on docs/to_go_further.md:
```diff
@@ -6,4 +6,4 @@
 Some papers have been published adressing such kind of questions [REF].
-[Small summary and an exemple of plot]
\ No newline at end of file
+[Small summary and an exemple of plot]
```
Why did you change that? Might cause conflict when merged with my PR... Could you rather delete that change and add a comment on my documentation PR if needed?

maylistran01 replied:
Ok, solved.

JulietteOrtholand reviewed (commented):
Thank you very much for this PR! The references and scientific contend is there, but I think this documentation could be improve by making a more explicit link with the code and how to actually do it :) Please let me know if my comments are not clear, and thanks again for this great work !

maylistran01 committed:
Correct citations and equations

maylistran01 committed:
Correct citations and equations

maylistran01 commented on docs/models_evaluation.md:
```diff
@@ -1,9 +1,123 @@
 
 ## Convergence diagnosis
 
+*We need to talk with Sofia in June to complete this part.*
+
+*At first glance, we would talk about looking at the convergence graphs saved in the logs (all the plots in convergence.pdf). WDYT?* 
```
Ok, I added it in the general presentation of the PR, in the TODO part and I added the link to the issue 107, thanks for pointing there !

JulietteOrtholand replied:
Maybe it is out of scope of this PR, WDYT?

maylistran01 replied:
You mean that we should merge before solving that in June, right? 
If so, yes I agree. I'll just keep it in the comment of the PR to keep it in mind for now. Or is there a better place to write this TODO to not forget, in github?

JulietteOrtholand replied:
I think this is the use of the issues. As documentation issue #358 contains everything, I would simply update it to precise that, WDYT?

maylistran01 replied:
Ok, I've updated the issue https://github.com/aramis-lab/leaspy/issues/107.

maylistran01 commented on docs/models_evaluation.md:
```diff
@@ -1,9 +1,123 @@
 ## Likelihood based metrics
 
-## Reconstruction metrics
+*Sofia and Gabrielle*
```
Done.

maylistran01 committed:
Apply suggestions from code review

Juliette first's suggestions

Co-authored-by: julietteortholand <juliette.ortholand@icm-institute.org>

maylistran01 reviewed (commented):
Thanks Juliette for the review ! 
I made the first easy-fixes. I'll come back to you when we've added the code with the application of Leaspy + computation of the metrics + the ref to the documentation of the package.

maylistran01 committed:
Remove sofia and gabrielle, add it to the PR comment

maylistran01 committed:
Remove $

maylistran01 commented on docs/models_evaluation.md:
```diff
@@ -1,9 +1,123 @@
+## Fit metrics
+
+In leaspy, 3 negative log likelihood (nll) are stored in the model’s json file, in the `fit_metrics` part:
+- `nll_attach`: Corresponds to the nll attach to the data
```
Ok, I've added the formula.

maylistran01 committed:
Add equation to likelihood

maylistran01 committed:
Add equation to likelihood

caglayantuna committed:
Docstrings algo (#370)

* algo docstrings

* add algorithm initialization

* Update src/leaspy/algo/settings.py

---------

Co-authored-by: Gensollen <nicolas.gensollen@gmail.com>

NicolasGensollen committed:
Clean dirty hacks (#371)

* Clean dirty hacks

* Remove another dirty hack

* Make things deterministic

* Update references for ARM

* Update references for not ARM

* small fix

JulietteOrtholand committed:
Documentation (#358)

* test markdown

* draft riemmanian

* change pdf for png

* image label

* drafft personnalise and litterature

* Riemmanian ok except ref

* algo missing code part2

* leaspy application

* Emilien suggestion

* push spell check

* mathematics with references

* test codespell

* test2

* test3

* test toml

* extra line toml

* delete change in spell-check

* code spell

* add ALS

* final version

* Apply suggestions from code review

Co-authored-by: caglayantuna <caglayan.tuna@icm-institute.org>

* update riemmanian link with application

* test ALS

* drop ALS

* update maylis ref

* new unseen

* modif Sofia

* Update docs/mathematics.md

Co-authored-by: KaisaridiSofia <115173948+KaisaridiSofia@users.noreply.github.com>

* changes sofia

* Apply suggestions from Nicolas code review

Co-authored-by: Gensollen <nicolas.gensollen@gmail.com>

---------

Co-authored-by: caglayantuna <caglayan.tuna@icm-institute.org>
Co-authored-by: KaisaridiSofia <115173948+KaisaridiSofia@users.noreply.github.com>
Co-authored-by: Gensollen <nicolas.gensollen@gmail.com>

GabrielleCasimiro committed:
Use `dataset.event_time_name` instead of hardcoded 'EVENT_TIME' (#376)

Co-authored-by: CASIMIRO Gabrielle <gabrielle.casimiro@umr-colli-mp021.icm-institute.org>

NicolasGensollen committed:
[DOC] Add examples gallery (#328)

* start working on examples gallery

* fix

* improve rendering

* try fixing weird error

* Better gallery header

* investigate failure

* fix image path

* try fixing build error

* try increasing timeout

* try increasing timeout in a different way

* try different

* clean conf and try fixing images

* fix path

* add missing images

* try things

* try things 2

* clean

* Apply suggestions from code review

Co-authored-by: julietteortholand <juliette.ortholand@icm-institute.org>

* fix user guide get_started page

---------

Co-authored-by: julietteortholand <juliette.ortholand@icm-institute.org>

caglayantuna committed:
Extend gitignore (#378)

NicolasGensollen committed:
Clean dirty hacks 2 (#374)

* clean another dirty hack

* check if randomness is the issue

* increase number of iterations

* decrease the number of iterations

NicolasGensollen committed:
Fix deprecations in `pyproject.toml` (#386)

* Update pyproject.toml to use new syntax and conventions

* do not spell check poetry.lock since it is auto generated

* Regenerate lock file

NicolasGensollen committed:
Add notations page (#377)

* try something for notations...

* Improve notations page and make some linking examples

* Update docs/notations.md

Co-authored-by: caglayantuna <caglayan.tuna@icm-institute.org>

* Apply suggestions from code review

Co-authored-by: julietteortholand <juliette.ortholand@icm-institute.org>

* Update docs/notations.md

---------

Co-authored-by: caglayantuna <caglayan.tuna@icm-institute.org>
Co-authored-by: julietteortholand <juliette.ortholand@icm-institute.org>

caglayantuna committed:
Docstring samplers (#388)

* improve docs of sampler

* remove hyper

NicolasGensollen committed:
Add some examples of warning, tip, and note blocks (#390)

NicolasGensollen committed:
Add references page (#391)

NicolasGensollen committed:
Clean another dirty hack (#389)

NicolasGensollen committed:
Merge univariate and multivariate models (#392)

* Merge univariate and multivariate (todo joint model)

* fixes

* more fixes

* update example

* merge joint and univariate_joint models

NicolasGensollen committed:
Rename multivariate parallel model (#394)

* rename multivariate parrallel model into share speed logistic model

* update tests

NicolasGensollen committed:
Refactor and rename `AbstractModel` and `AbstractMultivariateModel` (#395)

* move time reparam from abstract model to abstract multivariate model

* move method compute_individual_ages_from_biomarker_values from abstract model to abstract multivariate model

* move _audit_individual_parameters implem in abstract multivariate model

* mode put_individual_param implem into abstract multivariate model

* Refactor and rename models

NicolasGensollen committed:
Improve documentation configuration (#399)

* Clean examples in Makefile

* Update conf.py to work with GitHub

* Improve small things

NicolasGensollen committed:
Fix some documentation warnings relative to `leaspy.utils.typing` (#400)

* Fix documentation warnings relative to leaspy.utils.typing

* Update tests

NicolasGensollen committed:
Fix inheritance issues for algorithm methods `run` and `run_impl` (#402)

NicolasGensollen committed:
Propose a new architecture for algorithms (#403)

* Propose a new architecture for algorithms

* Remove working file

* Improve consistency of names

NicolasGensollen committed:
Rename models (#406)

NicolasGensollen committed:
Release `2.0.0rc1` (#407)

* update version number

* update changelog

* Change too optimistic date....

NicolasGensollen committed:
Remove `Leaspy` god class (#409)

* Remove Leaspy god class

* Update example

* Remove unused imports

* Refactor a few things

NicolasGensollen committed:
Merge `BaseModel` and `GenericModel` (#410)

NicolasGensollen committed:
Add `ModelInterface` (#411)

* Add ModelInterface

* some improvements

* some improvements again

NicolasGensollen committed:
Add `StatefulModel` (#413)

GabrielleCasimiro committed:
Covariate model - Load Data (#415)

* Update data.py dataset.py factory.py and individual_data.py, and add covariate_dataframe_data_reader.py to load covariates

* Update Dict/dit and List/list

* Apply suggestions from code review

Remove useless comments in covariate_dataframe_data_reader

Co-authored-by: Gensollen <nicolas.gensollen@gmail.com>

* Apply suggestions from code review

Remove useless comments

Co-authored-by: Gensollen <nicolas.gensollen@gmail.com>

* Update src/leaspy/io/data/covariate_dataframe_data_reader.py

Replace assertion by error raising

Co-authored-by: Gensollen <nicolas.gensollen@gmail.com>

* Apply suggestions to simplify conditions

---------

Co-authored-by: CASIMIRO Gabrielle <gabrielle.casimiro@umr-colli-mp021.icm-institute.org>
Co-authored-by: Gensollen <nicolas.gensollen@gmail.com>

maylistran01 committed:
[DOC] Documentation simulate (#368)

* plan proposition for the user guide

* Add doc for simulate

* Apply suggestions from code review

"Caglayan review"

Co-authored-by: caglayantuna <caglayan.tuna@icm-institute.org>

* Add Juliette's suggestions

* Better output table

* Add equations

* Add equations 2

* Adapt example to new version of simulate

* Adapt example to new version of simulate 2

---------

Co-authored-by: caglayantuna <caglayan.tuna@icm-institute.org>

lea-aguilhon committed:
Documentation joint model (#369)

maylistran01 committed:
Add code example for repeated measures prediction metrics

maylistran01 committed:
Resolve conflicts
